test/BreakEventHandler.js

```diff
@@ -0,0 +1,93 @@
+var expect = require('chai').expect,
+    launcher = require('./helpers/launcher'),
+    InjectorClient = require('../lib/InjectorClient').InjectorClient,
+    BreakEventHandler = require('../lib/BreakEventHandler').BreakEventHandler,
+    ScriptManager = require('../lib/ScriptManager').ScriptManager;
+    
+describe('BreakEventHandler', function() {
+  var childProcess,
+      session;
+
+  describe('should cache events', function() {
+    it('if started in debug-brk mode', function(done) {
+      prepareBreakEventHandler(true, function() {
+        expect(session.breakEventHandler._eventsCache).to.instanceof(Array);
+        expect(session.breakEventHandler._eventsCache.length).to.equal(0);
+        
+        session.debuggerClient.request('continue', undefined, function() {
+          childProcess.stdin.write('pause\n');
+          setTimeout(function() {
+            expect(session.breakEventHandler._eventsCache.length).to.equal(1);
+            done();
+          }, 100);
+        });
+      });
+    });
+    
+    it('if started in debug mode', function(done) {
+      prepareBreakEventHandler(false, function() {
+        expect(session.breakEventHandler._eventsCache).to.instanceof(Array);
+        expect(session.breakEventHandler._eventsCache.length).to.equal(0);
+        
+        childProcess.stdin.write('pause\n');
+        setTimeout(function() {
+          expect(session.breakEventHandler._eventsCache.length).to.equal(1);
+          done();
+        }, 100);
+      });
+    });
+  });
+  
+  describe('should resolve events', function() {
+    before(function(done) {
+      prepareBreakEventHandler(false, function() {
+        childProcess.stdin.write('pause\n');
+        setTimeout(done, 100);
+      });
+    });
+    
+    it('on resource-tree-resolved', function(done) {
+      session.frontendClient.once('Debugger.paused', function() {
+        done();
+      });
+      
+      session.debuggerClient.request('scripts', null, function(error, result) {
+        if (error) done(error);
+
+        result.forEach(session.scriptManager.addScript.bind(session.scriptManager));
+        session.resourceTreeResolved = true;
+        session.emit('resource-tree-resolved');
+      });
+    });
```

Comment:
So what is the failure message of this test when your fix is not applied?

#534 says

>  it seems that NodeJS 0.12 now signals the initial break as an actual breakpoint, and if you treat that as an unknown breakpoint and just instruct the debugger to continue, then the app immediately resumes and runs to completion.

> If I'm reading this correctly, their fix is to treat unknown breakpoints differently (basically; not instruct the debugger to continue on) while the debugger hasn't finished attaching itself.

Is there a way how to (re)write this unit-test to verify that the process was paused instead of running to completion? Or do I understand this wrong and the tests checks only BreakEvenHandler behaviour in isolation?

> BreakEventHandler  doesn't ignore events here

In that case you should be able to check that the event was triggered (sent to the client), is that correct? Is that what `session.frontendClient.once('Debugger.paused'` is doing?

Maybe it's starting to make more sense now. So your test is failing on a timeout before the fix is applied, because all breakpoints are ignored and the program will be never paused.  With the fix in place, the tests passed via `done` call from the event listener. Is that a correct description? Can you please add a comment to the test explaining this?